test/acceptance/test/core_operations_test.go

```diff
@@ -64,176 +65,314 @@ var (
+	if err != nil {
+		return err
+	}
+	client = c
+	return nil
 }
 
-func ensureWegoRepoIsAbsent(t *testing.T) {
+func ensureWegoRepoIsAbsent() error {
 	ctx := context.Background()
-	url := fmt.Sprintf("https://github.com/%s/%s", os.Getenv("GITHUB_ORG"), getWegoRepoName(t))
+	name, err := getWegoRepoName()
+	if err != nil {
+		return err
+	}
+	url := fmt.Sprintf("https://github.com/%s/%s", os.Getenv("GITHUB_ORG"), name)
 	ref, err := gitprovider.ParseOrgRepositoryURL(url)
-	require.NoError(t, err)
+	if err != nil {
+		return err
+	}
+
 	repo, err := client.OrgRepositories().Get(ctx, *ref)
 	if err != nil {
 		log.Info("Repo already deleted")
 	} else {
-		require.NoError(t, repo.Delete(ctx))
+		return repo.Delete(ctx)
 	}
 	clusterName, err := status.GetClusterName()
-	require.NoError(t, err)
+	if err != nil {
+		return err
+	}
 	repoName := clusterName + "-wego"
 	os.RemoveAll(fmt.Sprintf("%s/.wego/repositories/%s", os.Getenv("HOME"), repoName))
+	return nil
+}
+
+func ensureWegoRepoExists() error {
+	ctx := context.Background()
+	name, err := getRepoName()
+	if err != nil {
+		return err
+	}
+
+	url := fmt.Sprintf("https://github.com/%s/%s", os.Getenv("GITHUB_ORG"), name)
+	ref, err := gitprovider.ParseOrgRepositoryURL(url)
+	if err != nil {
+		return err
+	}
+	_, err = client.OrgRepositories().Get(ctx, *ref)
+	if err != nil {
+		return err
+	}
+	return nil
 }
 
-func ensureFluxVersion(t *testing.T) {
+func ensureFluxVersion() error {
 	if version.FluxVersion == "undefined" {
 		tomlpath, err := filepath.Abs("../../../tools/bin/stoml")
-		require.NoError(t, err)
+		if err != nil {
+			return err
+		}
 		deppath, err := filepath.Abs("../../../tools/dependencies.toml")
-		require.NoError(t, err)
+		if err != nil {
+			return err
+		}
 		out, err := utils.CallCommandSilently(fmt.Sprintf("%s %s flux.version", tomlpath, deppath))
-		require.NoError(t, err)
+		if err != nil {
+			return err
+		}
 		version.FluxVersion = strings.TrimRight(string(out), "\n")
 	}
+	return nil
 }
 
-func waitForNginxDeployment(t *testing.T) {
+func waitForNginxDeployment() error {
 	for i := 1; i < 61; i++ {
 		log.Infof("Waiting for nginx... try: %d of 60\n", i)
 		err := utils.CallCommandForEffect("kubectl get deployment nginx -n my-nginx")
 		if err == nil {
-			return
+			return err
 		}
 		time.Sleep(5 * time.Second)
 	}
-	require.FailNow(t, "Failed to deploy nginx workload to the cluster")
+	return fmt.Errorf("Failed to deploy nginx workload to the cluster")
 }
 
-func installFlux(t *testing.T) {
+func installFlux() error {
 	flux.SetupFluxBin()
 	manifests, err := fluxops.QuietInstall("wego-system")
-	require.NoError(t, err)
-	require.NoError(t, utils.CallCommandForEffectWithInputPipeAndDebug("kubectl apply -f -", string(manifests)))
+	if err != nil {
+		return err
+	}
+	return utils.CallCommandForEffectWithInputPipeAndDebug("kubectl apply -f -", string(manifests))
 }
 
-func getWegoRepoName(t *testing.T) string {
+func getWegoRepoName() (string, error) {
```

Comment:
Could either get rid of this entirely or just have it return `fluxops.GetRepoName()`